File: app/app/src/main/java/com/cabal/app/RatingFragment.java
package com.cabal.app;


import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.cabal.app.Utils.JsonLoader;

import java.util.Objects;


public class RatingFragment extends Fragment {

    public RatingFragment() {
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_rating, container, false);
        RecyclerView users = view.findViewById(R.id.recycleUsers);
        view.findViewById(R.id.rateOk).setOnClickListener(view1 -> {
            FragmentManager manager = Objects.requireNonNull(getActivity())
                .getSupportFragmentManager();
            manager.beginTransaction()
                .remove(this)
                .commit();
            manager.popBackStack();
                });
        users.setLayoutManager(new LinearLayoutManager(getContext()));
        RatingAdapter adapter = new RatingAdapter(JsonLoader.INSTANCE.loadUsers(getContext()));
        users.setAdapter(adapter);

        return view;
    }
}

File: app/app/src/main/java/com/cabal/app/RatingFragment.kt
package com.cabal.app

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.cabal.app.utils.JsonLoader

class RatingFragment : Fragment() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_rating, container, false)
        val users = view.findViewById<RecyclerView>(R.id.recycleUsers)
        view.findViewById<Button>(R.id.rateOk).setOnClickListener {
            activity?.supportFragmentManager?.run {
                beginTransaction()
                        .remove(this@RatingFragment)
                        .commit()
                popBackStack()
            }
        }
        users.run {
            layoutManager = LinearLayoutManager(context)
            adapter = RatingAdapter(JsonLoader.loadUsers(context))
        }
        return view
    }
}